core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -330,6 +362,15 @@ class FetchFailureHidingRDD(
       case t: Throwable =>
         if (throwOOM) {
           throw new OutOfMemoryError("OOM while handling another exception")
+        } else if (interrupt) {
+          // make sure our test is setup correctly
+          assert(TaskContext.get().asInstanceOf[TaskContextImpl].fetchFailed.isDefined)
+          // signal our test is ready for the task to get killed
```

Comment:
Maybe, `signal killingThread in our test`, since `killingThread` is truly waiting on `latch1`.

Reply:
I prefer the original comment -- the mechanics of what is waiting on the latch are easy enough to follow, its more important to explain why.